trade-finance-manager-api/src/v1/controllers/facility.controller.js

```diff
@@ -22,6 +27,66 @@ const getFacility = async (req, res) => {
   }
 };
 
+const getFacilities = async (req, res) => {
+  try {
+    const queryParams = req.body;
+
+    if (!queryParams) {
+      return {};
+    }
+
+    const dbFacilities = await api.getAllFacilities(queryParams);
+
+    const facilities = dbFacilities.map((dbFacility) => {
+      const { tfmFacilities: facility } = dbFacility;
+
+      if (!facility.ukefFacilityId) {
+        facility.ukefFacilityId = '-';
+      }
+
+      const latestCompletedAmendment = findLatestCompletedAmendment(dbFacility.amendments);
+      let facilityCoverEndDate = '';
+      let facilityCoverEndDateEpoch = '';
+
+      if (REGEX.DATE.test(facility.coverEndDate) && facility.coverEndDate) {
```

Comment:
Is this regex just the way it was done before? It also feels wrong to check facility.coverEndDate exists after testing it in regex before that
do you reckon we can do something instead where we create a new date from the coverEndDate and check that the date is valid after instead? (sorry I'm reviewing copy pasted code but worth having these noted)

Reply:
Ah yes, I did refactor this a bit, I've now updated the condition which doesn't make sense - [this](https://github.com/UK-Export-Finance/dtfs2/blob/main/trade-finance-manager-api/src/graphql/resolvers/query-facilities.js) is the code that I am replicating